video/image_writer.c

```diff
@@ -354,11 +371,34 @@ bool write_image(struct mp_image *image, const struct image_writer_opts *opts,
         destfmt = IMGFMT_RGB24;
     }
 #endif
+    if (opts->format == AV_CODEC_ID_WEBP && !opts->webp_lossless) {
+        // For lossy images, libwebp has its own RGB->YUV conversion.
+        // We don't want that, so force YUV/YUVA here.
```

Comment:
Why not? This would be just like libjpeg, which probably avoids subtle correctness issues, such as different chroma position.

Reply:
There's an annoying warning, that's it
https://github.com/FFmpeg/FFmpeg/blob/197985c5bf0e25fe15b93ceac0aec23edf3ec7ea/libavcodec/libwebpenc_common.c#L227-L230